src/genie/libs/parser/iosxe/show_interface.py

```diff
@@ -3220,42 +3220,60 @@ def cli(self, output=None):
+        # Po10      VSL LINK2          connected    trunk      a-full  a-10G
 
-        if out:
-            res = parsergen.oper_fill_tabular(device_output=out,
-                                              device_os='iosxe',
-                                              table_terminal_pattern=r"^\n",
-                                              header_fields=
-                                               [ "Port",
-                                                 "Name",
-                                                 "Status",
-                                                 "Vlan",
-                                                 "Duplex",
-                                                 "Speed",
-                                                 "Type"],
-                                              label_fields=
-                                              ["Interface",
-                                               "name",
-                                               "status",
-                                               "vlan",
-                                               "duplex_code",
-                                               "port_speed",
-                                               "type"],
-                                              index=[0]
-                                              )
-
-            #Building the schema out of the parsergen output
-            if res.entries:
-                 for intf, intf_dict in res.entries.items():
-                     intf = Common.convert_intf_name(intf)
-                     del intf_dict['Interface']
-                     parsed_dict.setdefault('interfaces', {}).update({intf: intf_dict})
 
-        return (parsed_dict)
+        # Name and Type are optional and also both can have multiple words on it, so find a single regex to match
+        # all cases is hard. I remove the Type column of the line and then the regex works for all cases
+        p1 = re.compile(r'^(?P<interfaces>[\d\w/]+)(?: +(?P<name>([\S ]+)))?'
+                        r' +(?P<status>\S+) +(?P<vlan>\S+) +(?P<duplex_code>[\S\-]+)'
+                        r' +(?P<port_speed>[\S\-]+)$')
+
+        type_pos = 0
+        for line in out.splitlines():
+            line = line.strip()
+
+            if "Type" in line:
+                type_pos = line.find("Type")
+                continue
+
+            if len(line) > type_pos:    # Type exists, lets get it and remove it
+                type_val = line[type_pos:].strip()
+                line = line[:type_pos -1]
+            else:
+                type_val = ""   # Type doesn't exists
+
+            m = p1.match(line)
+            if m:
+                group = m.groupdict()
+
+                intf_dict = result_dict.setdefault('interfaces', {}).\
+                                        setdefault(Common.convert_intf_name(group['interfaces']), {})
+
+                name_val = group['name'].strip()
```

Comment:
As I mentioned in another comment, if `type/name` is handled in regex, we can simply check their existence using:
```
group = m.groupdict()

keys = ['status', 'vlan', 'duplex_code', 'port_speed', 'name', 'type']

for k in keys:
  if group[k]:
   intf_dict[k] = group[k]
...
```